### PR TITLE
[Hub Generated] Review request for Microsoft.HybridCompute to add version preview/2021-01-28-preview

### PR DESCRIPTION
**Pull request by dorothyhu**
This is a PR generated at [OpenAPI Hub](https://aka.ms/openapihub). You can view your [work branch via this link](https://portal.azure-devex-tools.com/branch/dorothyhu/azure-rest-api-specs/dev-hybridcompute-Microsoft.HybridCompute-2021-01-28-preview...master/).

### Contribution checklist:
- [x] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of “no breaking changes
- [x] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [x] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist
- [x] Ensure to check this box if one of the following scenarios meet updates in the PR, so that label “WaitForARMFeedback” will be added automatically to involve ARM API Review. Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs, all “removals” and “adding a new property” no more require ARM API review.
  - Adding new API(s)
  - Adding a new API version
  - Adding a new service

- [ ] Please ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If there are following updates in the PR, ensure to request an approval from API Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 

- [ ] Removing API(s) in stable version
- [ ] Removing properties in stable version
- [ ] Removing API version(s) in stable version
- [ ] Updating API in stable version with Breaking Change Validation errors
- [ ] Updating API(s) in preview over 1 year

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

File: specification/hybridcompute/resource-manager/Microsoft.HybridCompute/preview/2021-01-28-preview/HybridCompute.json [hunk 1]
```diff
@@ -832,6 +832,17 @@
         "privateLinkScopeResourceId": {
           "type": "string",
           "description": "The resource id of the private link scope this machine is assigned to, if any."
+        },
+        "parentClusterResourceId": {
+          "type": "string",
+          "description": "The resource id of the parent cluster (Azure HCI) this machine is assigned to, if any."
+        },
+        "detectedProperties": {
+          "type": "object",
```
Comment (raghushantha) [inline specification/hybridcompute/resource-manager/Microsoft.HybridCompute/preview/2021-01-28-preview/HybridCompute.json:841]: Would be good to strongly type 'detectedProperties' and include as a field in the 'definitions' section of the swagger

Comment (dorothyhu) [reply specification/hybridcompute/resource-manager/Microsoft.HybridCompute/preview/2021-01-28-preview/HybridCompute.json:841]: can we strongly type dictionary<string, string> in the swagger?

Comment (raghushantha) [reply specification/hybridcompute/resource-manager/Microsoft.HybridCompute/preview/2021-01-28-preview/HybridCompute.json:841]: Hi @dorothyhu - you can create a new type. Look at Extensions as an example: $ref can point to a type

  {
            "name": "extensionParameters",
            "in": "body",
            "required": true,
            "schema": {
              "$ref": "#/definitions/MachineExtensionUpdate"
            },
            "description": "Parameters supplied to the Create Machine Extension operation."
          },

**Comment by dkirby017** [comment]
@ruowan Could you help us understand why this is flagged as a breaking change? All the changes are on our latest preview api version, and all the items in the checklist at the top refer to stable versions.

**Review by dkirby017 — APPROVED** [review]

**Comment by ruowan** [comment]
Adding properties in a public preview version, it's a breaking change.  For detail reason pls take a look at this comment https://github.com/Azure/azure-rest-api-specs/pull/13451#issuecomment-800473643.

**Comment by dkirby017** [comment]
@ruowan I'm afraid I don't understand why it's a breaking change, based on the documentation at the top of the PR. That checklist explicitly lists a whole bunch of changes to stable versions, or to a preview version over a year old. What is the point of having a preview version if we can't update it as the usage is fine-tuned? Might as well just make every version "stable"

**Review by raghushantha — APPROVED** [review]

**Comment by raghushantha** [comment]
@ruowan  - Adding to above response/observation.
We are planning to use this preview version as a staging ground for new features and additions and plan to release a stable version based on this.
It would be process heavy to create a new preview version for every one of these changes, since we need to update not just the Swagger, but also examples and associated documentation. We would also need to update our Customers with the new preview versions often, adding to the burden.
Also, the features that are being added will be behind a feature flag and available to a select set of Customers (if that helps contain the notion of introducing breaking changes)

**Comment by akning-ms** [comment]
Hi @raghushantha, latest breaking change policy is in this document: http://aka.ms/bcforapi which is first item in "Contribution checklist"
If you think you have strong justification to add new optional properties, you can ask for breaking change review board for approval follow below step: 
Action: To initiate an evaluation of the breaking change, create a new intake using the template for breaking changes. Addition details on the process and office hours are on the Breaking change Wiki.

**Comment by ruowan** [comment]
For private swagger repo, it only for private preview feature.  Public repo preview folder is for public preview feature.

Synced with @dorothyhu on teams, I believe he has already created an breaking change review intake work item.

**Comment by JeffreyRichter** [comment]
Azure's policy is that an api-version represents an API/swagger contract. So, if the API/swagger contract changes, the api-version MUST also change. For the reasons as to why this policy exists, please watch this video: https://youtu.be/bG984_L6dS0?list=PL9XzOCngAkqs4m0XdULJu_78nM3Ok3Q65&t=1135.

This holds true even when adding additional properties or operations (for the reasons mentioned in the video).
So, to remove the breaking change issues with this PR, please change the api-version.

**Comment by dorothyhu** [comment]
abandoned. I created a new API version for the change

**Pull request closed without merging**